mlir/tools/mlir-tblgen/OpPythonBindingGen.cpp

```diff
@@ -982,11 +1032,12 @@ static void emitOpBindings(const Operator &op, raw_ostream &os) {
   }
 
   emitRegionAttributes(op, os);
-  emitDefaultOpBuilder(op, os);
+  llvm::SmallVector<std::string> functionArgs = emitDefaultOpBuilder(op, os);
   emitOperandAccessors(op, os);
   emitAttributeAccessors(op, os);
   emitResultAccessors(op, os);
   emitRegionAccessors(op, os);
+  emitValueBuilder(op, functionArgs, os);
```

Comment:
Any reason to not just invoke the value builder emitter inside the op builder one?

Reply:
The value builder needs to defined at top-level (module) scope.